common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManagerSimpleCacheImpl.java

```diff
@@ -0,0 +1,196 @@
+ * store's API does not surface a success indicator. If you need stronger guarantees that an
+ * operation was successful, use {@link SharedPreferencesSimpleCacheImpl} which is less performance
+ * oriented.
+ * <p>
+ * Data serializes as JSON.
+ *
+ * @param <T> The type of metadata that will be persisted.
+ * @see SharedPreferencesSimpleCacheImpl
+ */
+public abstract class SharedPreferencesFileManagerSimpleCacheImpl<T> implements ISimpleCache<T>, IListTypeToken {
+
+    private static final String TAG = SharedPreferencesFileManagerSimpleCacheImpl.class.getSimpleName();
+    private static final String EMTPY_ARRAY = "[]";
+    private static final String TIMING_TAG = "execWithTiming";
+
+    private final ISharedPreferencesFileManager mSharedPrefsFileManager;
+    private final String mKeySingleEntry;
+    private final Gson mGson = new Gson();
+
+    /**
+     * Constructs a new SharedPreferencesFileManagerSimpleCacheImpl. Convenience class for persisting
+     * lists of arbitrarily-typed data.
+     *
+     * @param context   The current app's {@link Context}.
+     * @param prefsName The name of the underlying {@link android.content.SharedPreferences} file.
+     * @param singleKey The name of key under which all entries will be cached.
+     */
+    public SharedPreferencesFileManagerSimpleCacheImpl(@NonNull final Context context,
+                                                       @NonNull final String prefsName,
+                                                       @NonNull final String singleKey) {
+        Logger.verbose(
+                TAG + "::ctor",
+                "Init"
+        );
+        mSharedPrefsFileManager = SharedPreferencesFileManager.getSharedPreferences(
+                context,
+                prefsName,
+                Context.MODE_PRIVATE,
+                null // File is not encrypted
+        );
+        mKeySingleEntry = singleKey;
+    }
+
+    private interface NamedRunnable<V> extends Callable<V> {
+        String getName();
+    }
+
+    private <V> V execWithTiming(@NonNull final NamedRunnable<V> runnable) {
+        final long startTime;
+        final long execTime;
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+            startTime = SystemClock.elapsedRealtimeNanos();
+        } else {
+            startTime = System.nanoTime();
+        }
+
+        V v = null;
+        try {
+            v = runnable.call();
+        } catch (final Exception e) {
+            Logger.error(TAG + TIMING_TAG, "Error during operation", e);
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+            execTime = SystemClock.elapsedRealtimeNanos() - startTime;
+        } else {
+            execTime = System.nanoTime() - startTime;
+        }
+
+        Logger.verbose(TAG + TIMING_TAG,
+                runnable.getName() + " finished in: " + execTime + " " + TimeUnit.NANOSECONDS.name());
```

Comment:
can we put all of this in a finally block, so we get timing even if there's an exception?

Reply:
c14121ac4462e3f2a96ce2020cd47527e8a02d28